# Multiple models - unscaled

In [1]:
import numpy as np
import pandas as pd

from matplotlib import pyplot as plt 
import seaborn as sns

import cufflinks as cf

In [2]:
#cf.go_offline()

In [3]:
titanic = sns.load_dataset('titanic')

In [7]:
titanic.isnull().sum()

In [5]:
titanic.drop(columns=['deck'], inplace=True)
titanic.age = titanic.age.fillna(titanic.age.mean())
titanic.dropna(inplace=True)

In [8]:
unlabelled  = titanic.drop(['survived', 'alive'], axis=1)

In [9]:
unlabelled = pd.get_dummies(unlabelled, drop_first=True)

In [11]:
#unlabelled

In [56]:
#scaled = scaler.transform(unlabelled)

In [57]:
#xscaled = pd.DataFrame(scaled)

In [59]:
xscaled

In [13]:
from sklearn.model_selection import train_test_split

In [14]:
feat_train, feat_test, lab_train, lab_test = train_test_split(unlabelled, titanic.survived, test_size=0.2, random_state=101)

In [15]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from xgboost import XGBClassifier

In [16]:
knn = KNeighborsClassifier(n_neighbors=7)
knn.fit(feat_train, lab_train)
kpred = knn.predict(feat_test)

In [17]:
dtc = DecisionTreeClassifier()
dtc.fit(feat_train, lab_train)
dpred = dtc.predict(feat_test)

In [18]:
svc = SVC()
svc.fit(feat_train, lab_train)
spred = svc.predict(feat_test)

In [19]:
rfc = RandomForestClassifier(n_estimators=500)
rfc.fit(feat_train, lab_train)
rpred = rfc.predict(feat_test)

In [23]:
logit = LogisticRegression(n_jobs=1000)
logit.fit(feat_train, lab_train)
lpred = logit.predict(feat_test)

/Users/sikpe/opt/anaconda3/envs/cobra/lib/python3.9/site-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [24]:
xgb = XGBClassifier()
xgb.fit(feat_train, lab_train)
xpred = xgb.predict(feat_test)

In [25]:
from sklearn.metrics import classification_report

In [26]:
print(classification_report(kpred, lab_test))

              precision    recall  f1-score   support

           0       0.80      0.78      0.79       110
           1       0.66      0.69      0.68        68

    accuracy                           0.75       178
   macro avg       0.73      0.74      0.73       178
weighted avg       0.75      0.75      0.75       178



In [27]:
print(classification_report(spred, lab_test))

              precision    recall  f1-score   support

           0       0.90      0.68      0.77       141
           1       0.37      0.70      0.48        37

    accuracy                           0.69       178
   macro avg       0.63      0.69      0.63       178
weighted avg       0.79      0.69      0.71       178



In [28]:
print(classification_report(dpred, lab_test))

              precision    recall  f1-score   support

           0       0.82      0.79      0.81       111
           1       0.68      0.72      0.70        67

    accuracy                           0.76       178
   macro avg       0.75      0.75      0.75       178
weighted avg       0.77      0.76      0.77       178



In [29]:
print(classification_report(rpred, lab_test))

              precision    recall  f1-score   support

           0       0.89      0.83      0.86       115
           1       0.72      0.81      0.76        63

    accuracy                           0.82       178
   macro avg       0.80      0.82      0.81       178
weighted avg       0.83      0.82      0.82       178



In [30]:
print(classification_report(lpred, lab_test))

              precision    recall  f1-score   support

           0       0.92      0.82      0.86       120
           1       0.69      0.84      0.76        58

    accuracy                           0.83       178
   macro avg       0.80      0.83      0.81       178
weighted avg       0.84      0.83      0.83       178



In [31]:
print(classification_report(xpred, lab_test))

              precision    recall  f1-score   support

           0       0.88      0.84      0.86       112
           1       0.75      0.80      0.77        66

    accuracy                           0.83       178
   macro avg       0.81      0.82      0.82       178
weighted avg       0.83      0.83      0.83       178



# Performance
1. Logistic regression - 83%
2. XGBoot - 83%
3. Random Forest - 82%
4. Decision tree 76% 
5. K-Nearest Neighbor - 75%
6. Support Vector Machine 69%